src/DotLiquid/Util/ObjectExtensionMethods.cs

```diff
@@ -109,6 +109,20 @@ public static bool SafeTypeInsensitiveEqual(this object value, object otherValue
                 return false;
             }
 
+            // NOTE(Rodney Richardson): If either value is boolean or the string "true" or "false", compare with Falsy/Truthy
+            // We can't use IsTruthy() directly, as everything that's not Falsy is considered Truthy
+            bool valueIsFalsy = value.IsFalsy();
+            bool valueIsTruthy = (value is bool boolValue && boolValue == true)
+                || (value is string stringValue && "true".Equals(stringValue, StringComparison.OrdinalIgnoreCase));
+            bool otherValueIsFalsy = otherValue.IsFalsy();
+            bool otherValueIsTruthy = (otherValue is bool boolOtherValue && boolOtherValue == true)
+                || (otherValue is string stringOtherValue && "true".Equals(stringOtherValue, StringComparison.OrdinalIgnoreCase));
+
+            if (valueIsFalsy) return otherValueIsFalsy;

```

Comment:
Please observe the code formatting convention of the rest of the file - explicit code block for if conditions.

```suggestion
            if (valueIsFalsy)
            {
                return otherValueIsFalsy;
            }
```

etc.

Reply:
I had it like that, then shrank it down because it looked too bulky! :)